File: ruby/src/azure/mgmt/microsoft_compute/2016_01_31/models/virtual_machine.rb
require_relative "../../models/virtual_machine"

module Azure
  module Mgmt
    module Compute
      module Api_2016_01_31
        module Models
          class VirtualMachine < Azure::Mgmt::Compute::Models::VirtualMachine
            attr_accessor :vm_id

            def initialize(name:, location:, tags: nil, vm_id: nil)
              super(name: name, location: location, tags: tags)
              self.vm_id = vm_id
            end
          end
        end
      end
    end
  end
end

File: dotnet/src/Azure/Mgmt/Microsoft.Compute/2016_06_30/Models/VirtualMachine.cs
using System;
using System.Collections.Generic;
using Azure.Mgmt.Compute._2016_01_31.Models;
using Azure.Mgmt.Models;

namespace Azure.Mgmt.Compute._2016_06_30.Models
{
    public interface IVirtualMachine : IResource {
        string VmId { get; set; }
        IPlan Plan {get; set; }
        string LicenseType { get; set; }
        string ProvisioningState { get; }
    }
    public class VirtualMachine : Azure.Mgmt.Models.Resource, IVirtualMachine
    {
        public VirtualMachine(string name, string location, IDictionary<string, string> tags = null, string licenseType = null, string vmId = null, IPlan plan = null) : base(name, location, tags)
        {
            VmId = vmId;
            LicenseType = licenseType;
            Plan = plan;
        }

        /// <summary>
        /// Gets specifies the VM unique ID which is a 128-bits identifier that
        /// is encoded and stored in all Azure IaaS VMs SMBIOS and can be read
        /// using platform BIOS commands.
        /// </summary>
        public string VmId { get; set; }

        public IPlan Plan { get; set; }
        public string LicenseType { get; set; }
        public string ProvisioningState {get;}
    }
}